src/api/course/index.ts

```diff
@@ -320,13 +321,38 @@ export class CourseApi {
     return _.get((this.app.signalk as any).self, 'navigation.position')
   }
 
-  private validateCourseInfo(info: CourseInfo) {
+  private async validateCourseInfo(info: CourseInfo) {
     if (
       typeof info.activeRoute !== 'undefined' &&
       typeof info.nextPoint !== 'undefined' &&
       typeof info.previousPoint !== 'undefined'
     ) {
-      return info
```

Comment:
Previously this returned the `info` parameter, but now it is returning that in some cases and in others it is returning this.courseInfo. Why?

Reply:
If the stored state is deemed invalid it returns a valid default value.